core/src/main/java/org/elasticsearch/index/query/ExistsQueryBuilder.java

```diff
@@ -82,38 +80,17 @@ public static Query newFilter(QueryShardContext context, String fieldPattern) {
             return Queries.newMatchNoDocsQuery();
         }
 
-        ObjectMapper objectMapper = context.getObjectMapper(fieldPattern);
-        if (objectMapper != null) {
-            // automatic make the object mapper pattern
-            fieldPattern = fieldPattern + ".*";
-        }
-
-        Collection<String> fields = context.simpleMatchToIndexNames(fieldPattern);
-        if (fields.isEmpty()) {
-            // no fields exists, so we should not match anything
-            return Queries.newMatchNoDocsQuery();
+        final Collection<String> fields;
+        if (context.getObjectMapper(fieldPattern) != null) {
+            // the _field_names field also indexes objects, so we are good
```

Comment:
s/so we are good/so we don't have to do any more work to support exists queries on whole objects/ ?